### PR TITLE
Adding .expand method for TransformedDistribution

### PR DESCRIPTION
**Pull request by neerajprad**
This PR:
 - adds a `.expand` method for `TransformedDistribution` along the lines of #11341.
 - uses this method to simplify `.expand` in distribution classes that subclass off of `TransformedDistribution`.
 - restores testing of `TransformedDistribution` fixtures.
 - fixes some bugs wherein we were not setting certain attributes in the expanded instances, and adds tests for `.mean` and `.variance` which use these attributes.

### Motivation:

There are many cases where users directly use `TransformedDistribution` rather than subclassing off it. In such cases, it seems rather inconvenient to have to write a separate class just to define a `.expand` method. The default implementation should suffice in these cases.

cc. @fritzo, @vishwakftw, @alicanb

### CODE REVIEW

**Review by fritzo — APPROVED** [review]
LGTM, nice simplification! Feel free to ignore or address my nit as you like.

File: torch/distributions/relaxed_bernoulli.py [hunk 1]
```diff
@@ -120,10 +120,8 @@ def __init__(self, temperature, probs=None, logits=None, validate_args=None):
 
     def expand(self, batch_shape, _instance=None):
         new = self._get_checked_instance(RelaxedBernoulli, _instance)
-        base_dist = self.base_dist.expand(batch_shape)
-        super(RelaxedBernoulli, new).__init__(base_dist, SigmoidTransform(), validate_args=False)
-        new._validate_args = self._validate_args
-        return new
+        batch_shape = torch.Size(batch_shape)
```
[the remark this answers is not in the document]
Comment (neerajprad) [reply torch/distributions/relaxed_bernoulli.py]: I wouldn't miss a chance to delete some code! :)

**Review by fritzo — APPROVED** [review]
Looks great!

**Comment by neerajprad** [comment]
Should be good to merge.

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**